### PR TITLE
Move artifacts api mapping from urlrewrite to spring (done)

### PR DESCRIPTION
**Pull request by akshaymayekar**
1. Replaced RequestParams with PathVariables
2. Fixed test cases.

### CODE REVIEW

**Comment by AksMKR** [comment]
To be merged after urlrewrite PR is merged

File: server/src/main/java/com/thoughtworks/go/server/controller/ArtifactsController.java [hunk 1]
```diff
@@ -91,73 +97,63 @@
...
+    public ModelAndView getArtifacts(@PathVariable("pipelineName") String pipelineName,
+                                     @PathVariable("pipelineLabel") String counterOrLabel,
+                                     @PathVariable("stageName") String stageName,
+                                     @PathVariable(value = "stageCounter", required = false) String stageCounter,
+                                     @PathVariable("buildName") String buildName,
+                                     @RequestParam(value = "sha1", required = false) String sha,
+                                     @RequestParam(value = "serverAlias", required = false) String serverAlias,
+                                     @RequestParam(value = "startLineNumber", required = false) Long start,
+                                     HttpServletRequest request) throws Exception {
+        if (start == null) {
```
Comment (bdpiparva) [inline server/src/main/java/com/thoughtworks/go/server/controller/ArtifactsController.java]: in earlier implementation, it was done using filename `consoleout.json`. I think is better to keep it that way only.

File: server/src/main/java/com/thoughtworks/go/server/controller/ArtifactsController.java [hunk 2]
```diff
@@ -188,44 +184,19 @@ public ModelAndView postArtifact(@RequestParam("pipelineName") String pipelineNa
...
+    @RequestMapping(value = {"/files/{pipelineName}/{pipelineLabel}/{stageName}/{stageCounter}/{buildName}/*",
+            "/files/{pipelineName}/{pipelineLabel}/{stageName}/{stageCounter}/{buildName}/**"}, method = RequestMethod.PUT)
+    public ModelAndView putArtifact(@PathVariable("pipelineName") String pipelineName,
+                                    @PathVariable("pipelineLabel") String counterOrLabel,
+                                    @PathVariable("stageName") String stageName,
+                                    @PathVariable(value = "stageCounter", required = false) String stageCounter,
+                                    @PathVariable("buildName") String buildName,
                                     @RequestParam(value = "buildId", required = false) Long buildId,
-                                    @RequestParam("filePath") String filePath,
                                     @RequestParam(value = "agentId", required = false) String agentId,
```
[the remark this answers is not in the document]
Comment (bdpiparva) [reply server/src/main/java/com/thoughtworks/go/server/controller/ArtifactsController.java]: I don't think so it is needed now. It was used to update last heard time of agent.

**Comment by maheshp** [comment]
We need to re-look at how special characters are handled in path param. Closing this for now.

**Pull request closed without merging**